src/config.c

```diff
@@ -246,69 +228,50 @@ bud_config_t* bud_config_load(const char* path, int inlined, bud_error_t* err) {
   JSON_Object* log;
   JSON_Object* avail;
   JSON_Array* contexts;
-  bud_config_t* config;
 
-  str_from_file = NULL;
+  if (config->piped) {
+    char* content;
 
-  if (path == NULL) {
-    *err = bud_read_file_by_fd(0, &str_from_file);
-    if (!bud_is_ok(*err)) {
+    err = bud_read_file_by_fd(0, &content);
+    if (!bud_is_ok(err))
+      goto end;
+
```

Comment:
static char *config_key = "!config"; // You could set this as a global
size_t config_keylen = strlen(config_key); // Set this somewhere to be used for all

err = bud_hashmap_insert(&config->files.hashmap,
                                            config_key,
                                            config_keylen,
                                            (void*) content);
This should cut out hardcoding of "!config" which is used in numerous places